### PR TITLE
[New Utility] Workflow to rename FASTQ files (non-destructive)

### LINKED ISSUE #266: [Utilities] Add workflow to rename FASTQ files in a non-destructive manner
:cool:

### :pushpin: Explain the Request

For GISAID compatibility:

> To ensure that the FASTQ files are linked correctly, please rename the FASTQ files with the associated EPI_ISL_xxx ID OR Drag & Drop a .txt file containing a list of FASTQ filenames and corresponding FASTA sequence (EPI_ISL_xxx) together with your submission.

To facilitate the process on Terra, a new utility workflow needs to be created implementing the first suggestion on the quote.

### PR DESCRIPTION
**Pull request by cimendes**
Closes #266

## :hammer_and_wrench: Changes Being Made

This PR implements a new utility workflow as requested in #266. This workflow received a read file or a pair of read files (FASTQ), compressed or uncompressed, and returned a new, renamed and compressed FASTQ file for submission in GISAID. 

### Impacted Workflows/Tasks

None, this is a new implementation

## :brain: Context and Rationale

Requested by SFPHL

## :clipboard: Workflow/Task Steps

This is a sample-level workflow. If a reverse read (`read2`) is provided, the files get renamed to the provided `new_filename` input with the notation `<new_filename>_R1.fastq.gz` and `<new_filename>_R2.fastq.gz`. If only `read1` is provided, the file gets renamed to `<new_filename>.fastq.gz`. If a not-compressed file is provided, this gets compressed automatically by the workflow. ´

### Inputs

```
  input {
    File read1  
    File? read2
    String new_filename
  }
```

`read1`: Mandatory input; Forward-facing or single-end reads, compressed or uncompressed
`read2`: Optional input; Reverse-facing reads, compressed or uncompressed
`new_filename`: Mandatory input; String with new name for read files

### Outputs

```
output {
    String rename_fastq_files_version = version_capture.phb_version
    String rename_fastq_files_analysis_date = version_capture.date
    File read1_renamed = select_first([rename_PE_files.read1_renamed, rename_SE_files.read1_renamed])
    File? read2_renamed = rename_PE_files.read2_renamed
  }
```

`rename_fastq_files_version`: version of PHB used to run this workflow
`rename_fastq_files_analysis_date`: date of renaming
`read1_renamed`: Forward-facing or single-end reads. Always present
`read2_renamed`: Reverse-facing reads. Only present if `read2` is provided

### Impacted Outputs

None, new workflow

### Locally

Paired-end reads (compressed):
image

Single-end reads (compressed):
image

Single-end reads (uncompressed):
image

### Terra

~~Underway~~

22 Samples PE, Compressed: https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/ff1cda0d-1c92-4fc0-8b5e-7001d520cfe6

### Scenarios for Reviewer to Test

- Weird filenames
- Should we keep the `_R1.fastq.gz` output name or change it to something else?

## :microscope: Quality checks

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The workflow/task has been tested locally and on Terra
- [x] The CI/CD has been adjusted and tests are passing
- [x] Everything follows the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

### CODE REVIEW

**Review by emily-smith1 — APPROVED** [review]
Tested **successfully** on 10 single-end and 10 paired-end SARS-CoV-2 read files in the renaming_test table in [this Terra workspace](https://app.terra.bio/#workspaces/cdph-terrabio-taborda-manual/CDPH_Smith_Sandbox/data). Files were renamed according to corresponding GISAID accession per the original request.

File: tasks/utilities/task_rename_files.wdl [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  input {
+    File read1
+    File? read2
+    String new_filename
+    String docker = "us-docker.pkg.dev/general-theiagen/ubuntu/ubuntu:jammy-20230816"
+    Int disk_size = 100
+    Int cpu = 2
+    Int mem = 2
+  }
+  command <<<
```
Comment (sage-wright) [inline tasks/utilities/task_rename_files.wdl:16]: do we need to have two separate tasks for this? the way the PE  task is set up looks like we shouldn't need the SE task?

**Review by sage-wright — CHANGES_REQUESTED** [review]
Looks good, but would drop the single end version, since r2 is already optional and you're checking for its existence in the PE version of the task anyways.

**Comment by cimendes** [comment]
> Looks good, but would drop the single end version, since r2 is already optional and you're checking for its existence in the PE version of the task anyways.

The reason why I introduced the SE task was because I couldn't find a smart way to have the output be *.fastq.gz or *_1.fastq.gz for the forward read... Maybe we can think of something smart together? :)

**Pull request merged**